sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/KeyVaultECDsa.cs

```diff
@@ -0,0 +1,184 @@
+            KeySizeValue = _publicKey.KeySize;
+            LegalKeySizesValue = new[] { new KeySizes(_publicKey.KeySize, _publicKey.KeySize, 0) };
+        }
+
+        private void CheckDisposed()
+        {
+            if (_publicKey is null)
+                throw new ObjectDisposedException($"{nameof(KeyVaultECDsa)} is disposed.");
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _publicKey?.Dispose();
+                _publicKey = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        /// <inheritdoc />
+        public override byte[] SignHash(byte[] hash)
+        {
+            CheckDisposed();
+            ValidateKeyDigestCombination(KeySize, hash.Length);
+
+            // We know from ValidateKeyDigestCombination that the key size and hash size are matched up
+            // according to RFC 7518 Sect. 3.1.
+            if (KeySize == 256)
+                return Context.SignDigest(hash, HashAlgorithmName.SHA256, KeyVaultSignatureAlgorithm.ECDsa);
+            if (KeySize == 384)
+                return Context.SignDigest(hash, HashAlgorithmName.SHA384, KeyVaultSignatureAlgorithm.ECDsa);
+            if (KeySize == 521) //ES512 uses nistP521
+                return Context.SignDigest(hash, HashAlgorithmName.SHA512, KeyVaultSignatureAlgorithm.ECDsa);
+
+            throw new ArgumentException("Digest length is not valid for the key size.", nameof(hash));
+        }
+
+        /// <inheritdoc />
+        protected override byte[] HashData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
+        {
+            try
+            {
+                ValidateKeyDigestCombination(KeySize, hashAlgorithm);
+
+                using (IncrementalHash hash = IncrementalHash.CreateHash(hashAlgorithm))
+                {
+                    hash.AppendData(data, offset, count);
+                    return hash.GetHashAndReset();
+                }
+            }
+            catch (Exception e)
+            {
+                throw new CryptographicException("Could not hash the data", e);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override bool VerifyHash(byte[] hash, byte[] signature)
+        {
+            CheckDisposed();
+
+            try
+            {
+                ValidateKeyDigestCombination(KeySize, hash.Length);
+            }
+            catch (Exception e)
+            {
+                throw new CryptographicException("Could not verify hash.", e);
+            }
+
+            return _publicKey.VerifyHash(hash, signature);
+        }
+
+        ///<inheritdoc/>
+        public override ECParameters ExportParameters(bool includePrivateParameters)
+        {
+            if (includePrivateParameters)
+                throw new CryptographicException("Private keys cannot be exported by this provider");
```

Comment:
Didn't see if you replied to the old comment, but this isn't strictly true. We also support creating a `CryptographyClient` from a JWK, which could have a private key (in fact, that's the main reason we did). So it could be exportable. I would just pass it through and let the underlying implementation throw as appropriate.